### PR TITLE
Add `PUT` endpoint for updating experience

### LINKED ISSUE #8: Edit existing Experience
Using a `PUT` request for the `/resume/experience` route, update an existing Experience using its index position as an ID. As part of this, you should write a test in `test_pytest.py` to show it works.

### PR DESCRIPTION
**Pull request by ArshiaZr**
Resolves #8 
**Description**:
- Added `PUT` `/resume/experience/{id}`:
  - Validates required fields (`title`, `company`, `start_date`, `end_date`, `description`).
  - Optionally handles `logo` file.
  - Returns 400 for missing/invalid fields.
  - On success, returns a message and updated record ID.

**Tests**:
- Added `test_update_experience` to verify:
  - Invalid experience IDs return 400.
  - Successful update of experience returns 200.

### CODE REVIEW

**Comment by ArshiaZr** [comment]
@Pradyuman7

**Comment by ArshiaZr** [comment]
One test fails which is another part of the code that has been written by someone else I can also resolve it @Pradyuman7

File: app.py [hunk 1]
```diff
@@ -143,28 +143,90 @@ def experience():
...
+        data["experience"][index] = Experience(
+            request_body["title"],
+            request_body["company"],
+            request_body["start_date"],
+            request_body["end_date"],
+            request_body["description"],
+            logo_filename,
+        )
+
+        return jsonify({"message": "Experience updated", "id": index}), 200
```
Comment (Pradyuman7) [inline app.py:228]: nit: Maybe 204 would be better here, since this updates the resource only

File: app.py [hunk 2]
```diff
@@ -143,28 +143,90 @@ def experience():
...
+        missing_fields = [
+            field for field in required_fields if field not in request_body
+        ]
+        invalid_fields = [
+            field
+            for field, field_type in required_fields.items()
+            if field in request_body and not isinstance(request_body[field], field_type)
+        ]
+
+        if missing_fields or invalid_fields:
```
Comment (Pradyuman7) [inline app.py:199]: I believe there's a relevant method in helpers for this which you can use

**Comment by ArshiaZr** [comment]
Fixed them @Pradyuman7

**Review by Pradyuman7 — APPROVED** [review]
LGTM 💪 
Please resolve the conflicts and then I'll merge it

**Comment by ArshiaZr** [comment]
@Pradyuman7  resolved

**Pull request merged**